src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -476,10 +476,24 @@ public function __clone()
      */
     public function __toString()
     {
+        $protocol = '';
+        try {
+            $protocol = $this->server->get('SERVER_PROTOCOL');
+        } catch (\InvalidArgumentException $mallformedPath) {
+            trigger_error($mallformedPath->getMessage(), E_USER_ERROR);
+        }
+
+        $content = '';
+        try {
+            $content = $this->getContent();
+        } catch (\LogicException $usageRestrictions) {
+            trigger_error($usageRestrictions->getMessage(), E_USER_ERROR);
```

Comment:
I'm wrong on this one: it can happen if getContent() has be *previously* called with `true`.
I'd be OK to keep this one, but :-1: for the other.